libcaf_test/caf/test/runner.cpp

```diff
@@ -68,7 +68,21 @@ int runner::run(int argc, char** argv) {
     return EXIT_SUCCESS;
   }
   default_reporter->start();
+  auto enabled = [](const std::regex& selected,
+                    std::string_view search_string) {
+    return std::regex_search(search_string.begin(), search_string.end(),
+                             selected);
+  };
+  auto suite_regex = runner::to_regex(get_or(cfg_, "suites", ".*"));
+  if (!suite_regex) {
+    using detail::format_to;
+    auto err = std::ostream_iterator<char>{std::cerr};
+    format_to(err, "regex could not be parsed\n");
```

Comment:
As a user, I wouldn't know what this should tell me. The error should give the user context so that he can actually fix what's wrong, but this description omits the critical information which option this error belongs to. Is there a specific reason why you deviated from the wording we use in [`parse_cli`](https://github.com/actor-framework/actor-framework/blob/e4866c8638575bd912612acaecb3c6e71579dee8/libcaf_test/caf/test/runner.cpp#L131)?

Furthermore, this is a very generic error. By doing the error reporting here, we lost the context we had in `to_regex`. Specifically, the `what()` string from [`regex_error`](https://en.cppreference.com/w/cpp/regex/regex_error). Pushing the error reporting into the function would allow us to do better error reporting. Of course then the function needs the option name. Also, we'll add another option with a regex input (`-t`) that will need to do the same error reporting again.

Reply:
Are you referring to this format for reporting errors 
`error while parsing argument '{}': {}\n\n{}\n` ?